pkg/operator/controller/controller.go

```diff
@@ -149,6 +149,8 @@ func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 					// Handle deletion.
 					if err := r.ensureIngressDeleted(ingress, dnsConfig, infraConfig); err != nil {
 						errs = append(errs, fmt.Errorf("failed to ensure ingress deletion: %v", err))
+					} else if err := r.syncOperatorStatus(); err != nil {
```

Comment:
This doesn't seem to match the PR description. If `r.ensureIngressDeleted` on L150 returns an error, isn't `r.syncOperatorStatus` on L168 still called?